### PR TITLE
Capture of the applicant and ecologist names 

### PR DESCRIPTION
**Pull request by GrahamMWillis**
The start of the contacts journey (applicant and ecologist)

### CODE REVIEW

File: packages/powerapps-lib/src/schema/processors/__tests__/example-src-object.js [hunk 1]
```diff
@@ -329,10 +329,10 @@ export const updatedGeneratedAssignmentsObject = [
...
+export const expectedApplicationRequestPath = 'sdds_applications?$select=sdds_applicationnumber,sdds_descriptionofproposal,sdds_detailsofconvictions,sdds_whydoyouneedalicence,sdds_applicationcategory,sdds_isapplicantonwnerofland,sdds_receivedonwerpermission,sdds_doestheprojectneedanypermissions,sdds_projectpermissionsgranted&$filter=sdds_sourceremote eq true&$expand=sdds_applicantid($select=lastname,telephone1,emailaddress1,address1_line1,address1_line2,address1_line3,address1_county,address1_city,address1_postalcode),sdds_organisationid($select=name,telephone1,emailaddress1,address1_line1,address1_line2,address1_line3,address1_county,address1_city,address1_postalcode),sdds_ecologistid($select=lastname,telephone1,emailaddress1,address1_line1,address1_line2,address1_line3,address1_county,address1_city,address1_postalcode),sdds_ecologistorganisationid($select=name,telephone1,emailaddress1,address1_line1,address1_line2,address1_line3,address1_county,address1_city,address1_postalcode),sdds_applicationtypesid($select=sdds_applicationname,sdds_description,sdds_appsuffix),sdds_applicationpurpose($select=sdds_name,sdds_description)'
```
Comment (kealanparr) [inline packages/powerapps-lib/src/schema/processors/__tests__/example-src-object.js:332]: What is this?

File: packages/powerapps-lib/src/schema/processors/__tests__/example-src-object.js [hunk 2]
```diff
@@ -356,7 +356,7 @@ export const applicationResponseObject = {
   sdds_organisationid: null,
   sdds_ecologistid: {
     '@odata.etag': 'W/"3285316"',
-    fullname: 'Mr Brian Ecologist',
+    lastname: 'Mr Brian Ecologist',
```
Comment (kealanparr) [inline packages/powerapps-lib/src/schema/processors/__tests__/example-src-object.js:359]: Can we please update the value to better reflect the key?

File: packages/powerapps-lib/src/schema/processors/__tests__/schema-processes.spec.js [hunk 3]
```diff
@@ -50,7 +50,7 @@ describe('the schema processes', () => {
       expect(applicationPayload).toEqual({
         relationshipsPayload: null,
         columnPayload: {
-          fullname: 'Mr Brian Yak',
+          lastname: 'Mr Brian Yak',
```
Comment (kealanparr) [inline packages/powerapps-lib/src/schema/processors/__tests__/schema-processes.spec.js:53]: Can we please update the value to better reflect the key?

File: packages/powerapps-lib/src/schema/processors/__tests__/schema-processes.spec.js [hunk 4]
```diff
@@ -101,7 +101,8 @@ describe('the schema processes', () => {
             sdds_osgridreference: '7868962',
             sdds_addressline1: 'Winscombe',
             sdds_county: 'Somerset',
-            sdds_postcode: 'BS21 2XX'
+            sdds_postcode: 'BS21 2XX',
+            sdds_sourceremote: true
```
Comment (kealanparr) [inline packages/powerapps-lib/src/schema/processors/__tests__/schema-processes.spec.js:105]: What is this please?

File: packages/web-service/src/handlers/__tests__/submit-application.spec.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    const mockSubmit = jest.fn()
+    jest.doMock('../../services/api-requests.js', () => ({
+      APIRequests: {
+        APPLICATION: {
+          submit: mockSubmit
+        }
+      }
+    }))
+    const mockRedirect = jest.fn(() => 'next')
+    const h = {
```
Comment (kealanparr) [inline packages/web-service/src/handlers/__tests__/submit-application.spec.js:23]: Is this the response mock? `h` is a vague variable name

File: packages/web-service/src/pages/contact/common/common.js [hunk 6]
```diff
@@ -1,4 +1,23 @@
...
+  return APIRequests.USER.getById(userId)
+}
+
+export const getApplicantData = async request => {
+  const journeyData = await request.cache().getData()
+  const { userId, applicationId } = journeyData
+  return APIRequests.APPLICANT.getById(userId, applicationId)
+}
+
+export const getEcologistData = async request => {
```
Comment (kealanparr) [inline packages/web-service/src/pages/contact/common/common.js:16]: This seems like a duplicated function

Can we maybe have a base function and pass in what we need for the different versions of the function?

Maybe I've misunderstood, just curious!

**Review by kealanparr — COMMENTED** [review]
Just some questions and queries to help me better understand

**Review by sfaulkner89 — COMMENTED** [review]
Kealan has left some questions we had for you.

File: packages/powerapps-lib/src/schema/processors/__tests__/example-src-object.js (hunk 1 above)
Comment (GrahamMWillis) [reply packages/powerapps-lib/src/schema/processors/__tests__/example-src-object.js:332]: Its test data. This is the ODATA query created by applying the model i.e. 

`buildRequestPath(SddsApplication, [Contact, Account, SddsApplicationType, SddsApplicationPurpose])`

File: packages/powerapps-lib/src/schema/processors/__tests__/example-src-object.js (hunk 2 above)
Comment (GrahamMWillis) [reply packages/powerapps-lib/src/schema/processors/__tests__/example-src-object.js:359]: Names and title capture are currently under review. I am going to change this at the point there is an agreed way forward.

File: packages/powerapps-lib/src/schema/processors/__tests__/schema-processes.spec.js (hunk 3 above)
Comment (GrahamMWillis) [reply packages/powerapps-lib/src/schema/processors/__tests__/schema-processes.spec.js:53]: See above

File: packages/powerapps-lib/src/schema/processors/__tests__/schema-processes.spec.js (hunk 4 above)
Comment (GrahamMWillis) [reply packages/powerapps-lib/src/schema/processors/__tests__/schema-processes.spec.js:105]: We will go through this when we discuss the queue. The tests in this module are concerned with the various processes involved in the transformation of data from our internal format into an ODATA batch query.

File: packages/web-service/src/handlers/__tests__/submit-application.spec.js (hunk 5 above)
Comment (GrahamMWillis) [reply packages/web-service/src/handlers/__tests__/submit-application.spec.js:23]: 'h' is the generic name given to the hapi response toolkit. It is called 'h' in the documentation. Yes, it a poor name! responseToolkit would have been better, but I think it should align with the documentation. I am keen on canonical usage wherever possible.

File: packages/powerapps-lib/src/schema/processors/__tests__/example-src-object.js (hunk 2 above)
Comment (kealanparr) [reply packages/powerapps-lib/src/schema/processors/__tests__/example-src-object.js:359]: That makes sense - thanks for explaining

**Pull request merged**